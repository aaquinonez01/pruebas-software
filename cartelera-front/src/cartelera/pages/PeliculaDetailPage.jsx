import { Button, Card } from "@tremor/react";
import { useEffect, useState } from "react";
import { useNavigate, useParams } from "react-router-dom";
import { peliculaService } from "../../services/pelicula.service";

export const PeliculaDetailPage = () => {
  const [pelicula, setPelicula] = useState({});
  const { id } = useParams();
  const navigate = useNavigate();

  const handleDelete = (id) => {
    peliculaService
      .delete(id)
      .then((response) => {
        console.log(response);
        navigate("/peliculas");
      })
      .catch((error) => {
        console.log(error);
      });
  };
  const getPeliculaById = async () => {
    // Llamar al servicio de pelicula por id
    const response = await peliculaService.get(id);
    setPelicula(response.data);
  };

  useEffect(() => {
    if (id) {
      getPeliculaById();
    }
  }, [id]);

  return (
    <Card className="max-w-4xl mx-auto mt-3">
      <div className="flex space-x-4 p-4">
        <div className="w-1/3">
          <img
            src={pelicula.imagenURL}
            alt={pelicula.titulo}
            className="w-full h-auto"
          />
        </div>
        <div className="w-2/3">
          <p
            className="text-3xl font-bold text-titulo"
            data-testid="titulo-pelicula"
          >
            {pelicula.titulo}
          </p>
          <p
            className="text-gray-600 font-bold text-anio"
            data-testid="anio-pelicula"
          >
            Año: {pelicula.anio}
          </p>
          <p
            className="text-gray-600 font-bold text-categoria"
            data-testid="categoria-pelicula"
          >
            Categoría: {pelicula.categoria}
          </p>
          <p
            className="text-gray-600 text-descripcion"
            data-testid="descripcion-pelicula"
          >
            Descripción: {pelicula.descripcion}
          </p>
          {/* Agrega más información de la película según tu modelo de datos */}
          <Button
            variant="primary"
            className="bg-red-500 hover:bg-red-600 rounded-lg mt-4"
            onClick={() => handleDelete(pelicula.id)}
          >
            Eliminar Pelicula
          </Button>
        </div>
      </div>
    </Card>
  );
};
