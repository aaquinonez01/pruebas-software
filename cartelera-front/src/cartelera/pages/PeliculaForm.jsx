import { Button, Card, Text, TextInput, Title } from "@tremor/react";
import { useEffect, useState } from "react";
import { useNavigate } from "react-router-dom";
import { useForm } from "../../hooks/useForm";
import categoriaService from "../../services/categoria.service";
import { peliculaService } from "../../services/pelicula.service";

const PeliculaForm = () => {
  const { titulo, imagenURL, anio, descripcion, categoria, onInputChange, imagenBase64 } = useForm({
    titulo: "",
    imagenURL: "",
    anio: "",
    descripcion: "",
    categoria: "",
    imagenBase64: ""
  })

  const navigate = useNavigate();
  const [categorias, setCategorias] = useState([]);

  const getCategorias = () => {
    categoriaService
      .getAll()
      .then(response => {
        setCategorias(response);
      })
      .catch(error => {
        console.log(error);
      });
  };

  useEffect(() => {
    getCategorias();
  }, []);

  const handleSubmit = e => {
    e.preventDefault();
    peliculaService
      .create({
        titulo,
        anio,
        categoria,
        descripcion,
        imagenURL: imagenBase64,
      })
      .then(response => {
        console.log(response);
        navigate("/peliculas");
      })
      .catch(error => {
        console.log(error);
      });
    // Limpia los campos después de envia
  };

  return (
    <div className="flex justify-center items-center h-screen">
      <Card className="w-3/4 p-4 flex">
        <div className="w-1/2 mr-4">
          <Title className="text-center text-3xl font-bold">
            Agregar Nueva Película
          </Title>
          <form className="flex flex-col gap-4 mt-5" onSubmit={handleSubmit}>
            <Text>Título:</Text>
            <TextInput
              type="text"
              label="Título"
              value={titulo}
              onChange={onInputChange}
              name="titulo"
              required
            />
            <div className="flex gap-4 w-full">
              <div>
                <Text>Año de Estreno:</Text>
                <TextInput
                  type="number"
                  label="Año de Estreno"
                  value={anio}
                  onChange={onInputChange}
                  name="anio"
                  required
                />
              </div>
              <div className="w-full">
                <Text>Categoría:</Text>
                <select
                  label="Categoría"
                  value={categoria}
                  onChange={onInputChange}
                  name="categoria"
                  required
                  className="
                    w-full border border-gray-300 rounded px-3 py-2
                  "
                >
                  <option value="">Seleccione una categoría</option>
                  {categorias.map(categoria => (
                    <option key={categoria.id} value={categoria.id}>
                      {categoria.nombre}
                    </option>
                  ))}

                  {/* Agrega más opciones de categorías aquí */}
                </select>
              </div>
            </div>
            <Text className="relative">Descripción:</Text>
            <textarea
              className="border border-gray-300 rounded px-3 py-2 h-32"
              label="Descripción"
              value={descripcion}
              onChange={onInputChange}
              name="descripcion"
              required
            ></textarea>
            <Button
              type="submit"
              className="bg-blue-500 hover:bg-blue-600 text-white"
            >
              Agregar Película
            </Button>
          </form>
        </div>
        <div className="w-1/2 flex flex-col justify-center items-center ">
          <Card className="p-4 w-72 h-96 mx-auto">
            <img
              src={imagenURL}
              className="w-full h-full object-cover"
            />
          </Card>
          <TextInput
            className="w-1/2 mx-auto mt-4"
            type="file"
            label="Imagen"
            accept="image/*"
            name="imagenURL"
            onChange={onInputChange}
            required
          />
        </div>
      </Card>
    </div>
  );
};

export default PeliculaForm;
