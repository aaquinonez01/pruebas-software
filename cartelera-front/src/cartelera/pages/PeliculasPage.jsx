import { Badge, Card, Col, Grid } from "@tremor/react";
import { Link } from "react-router-dom";
import { usePelicula } from "../hooks/usePelicula";
const PeliculasPage = () => {
  const { peliculas, handleDelete } = usePelicula();


  return (
    <div className="p-4 mx-32 mt-6">
      <h1 className="text-3xl font-semibold mb-4 text-center">
        Cartelera de Películas
      </h1>
      <Grid gap={8} className="grid-cols-2 md:grid-cols-3 lg:grid-cols-4 mt-10">
        {peliculas.length > 0 &&
          peliculas.map(pelicula => (
            <Col key={pelicula.id} className="mb-4">
              <Card className="border-none">
                <img
                  src={pelicula.imagenURL} // Añade la URL de la imagen de la película
                  alt={pelicula.titulo}
                  className="w-full h-auto"
                />
                <div className="p-2">
                  <h2 className="text-lg font-semibold mb-2">
                    {pelicula.titulo}
                  </h2>
                  <p className="mb-2">{pelicula.anio}</p>
                  <div className="mb-2">
                    <Badge>{pelicula.categoria}</Badge>
                  </div>
                  <Link
                    to={`/peliculas/${pelicula.id}`}
                    className="w-full bg-yellow-500 hover:bg-yellow-600 text-white px-4 py-2"
                  >
                    Ver Detalles
                  </Link>
                </div>
              </Card>
            </Col>
          ))}
      </Grid>
    </div>
  );
};

export default PeliculasPage;
