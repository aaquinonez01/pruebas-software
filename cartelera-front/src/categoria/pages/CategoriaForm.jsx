import { Button, Card, TextInput, Title } from "@tremor/react"; // Importa los componentes necesarios
import { useEffect, useState } from "react";
import { useNavigate, useParams } from "react-router-dom";
import categoriaService from "../../services/categoria.service";
export const CategoriaForm = () => {
  const { id } = useParams();
  const [tipoAccion, setTipoAccion] = useState("Crear"); // Define el tipo de acción [Crear, Editar
  const [categoria, setCategoria] = useState({
    nombre: "",
    descripcion: "",
  }); // Define el estado inicial de la categoría
  const navigate = useNavigate();
  const handleChange = (event) => {
    setCategoria({
      ...categoria,
      [event.target.name]: event.target.value,
    });
  }; // Actualiza el estado de la categoría

  useEffect(() => {
    if (id) {
      setTipoAccion("Editar");
      categoriaService
        .get(id)
        .then((response) => {
          console.log(response);
          setCategoria(response);
        })
        .catch((error) => {
          console.log(error);
        });
    }
  }, [id]); // Llama al servicio de categoría por id

  const handleSubmit = (event) => {
    event.preventDefault();
    if (tipoAccion === "Crear") {
      categoriaService
        .create({
          nombre: categoria.nombre,
          descripcion: categoria.descripcion,
        })
        .then((response) => {
          console.log(response);
        })
        .catch((error) => {
          console.log(error);
        });
      navigate("/categorias");
    } else {
      categoriaService
        .update(id, {
          nombre: categoria.nombre,
          descripcion: categoria.descripcion,
        })
        .then((response) => {
          console.log(response);
        })
        .catch((error) => {
          console.log(error);
        });
      navigate("/categorias");
    }
  };
  return (
    <div>
      <Card>
        <Title className="text-center text-3xl">Crear Nueva Categoría</Title>
        <form
          className="mt-5 space-y-4 w-2/5 mx-auto"
          onSubmit={handleSubmit}
        >
          <div>
            <label className="block font-medium">Nombre:</label>
            <TextInput
              data-cy="nombre-categoria"
              type="text"
              placeholder="Ingrese el nombre de la categoría"
              className="w-full"
              name="nombre"
              value={categoria.nombre}
              onChange={handleChange}
            />
          </div>
          <div>
            <label className="block font-medium">Descripción:</label>
            <TextInput
              data-cy="descripcion-categoria"
              placeholder="Ingrese la descripción de la categoría"
              className="w-full"
              name="descripcion"
              value={categoria.descripcion}
              onChange={handleChange}
            />
          </div>
          <div className="flex justify-end">
            <Button
              className="bg-blue-500 hover:bg-blue-600 text-white px-4 py-2 rounded"
              data-cy="data-cy=boton-actualizar"
            >
              {tipoAccion === "Crear" ? "Crear Categoría" : "Editar Categoría"}
            </Button>
          </div>
        </form>
      </Card>
    </div>
  );
};
