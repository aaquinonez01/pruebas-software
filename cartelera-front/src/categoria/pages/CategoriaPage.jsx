import {
  Card,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableHeaderCell,
  TableRow,
  Title,
} from "@tremor/react";
import { useEffect, useState } from "react";
import { Link } from "react-router-dom";
import categoriaService from "../../services/categoria.service";

export const CategoriaPage = () => {
  const [categorias, setCategorias] = useState([]);
  const getCategorias = () => {
    categoriaService
      .getAll()
      .then((response) => {
        setCategorias(response);
      })
      .catch((error) => {
        console.log(error);
      });
  };
  const handleDelete = (id) => {
    categoriaService
      .delete(id)
      .then((response) => {
        console.log(response);
        getCategorias();
      })
      .catch((error) => {
        console.log(error);
      });
  };

  useEffect(() => {
    getCategorias();
  }, []);

  return (
    <div>
      <Card>
        <Title className="text-center text-3xl">Lista de Categor√≠as</Title>
        <Table className="mt-5 w-2/3 mx-auto">
          <TableHead className="text-center">
            <TableRow>
              <TableHeaderCell className="text-center">ID</TableHeaderCell>
              <TableHeaderCell className="text-center">NOMBRE</TableHeaderCell>
              <TableHeaderCell className="text-center">
                DESCRIPCION
              </TableHeaderCell>
              <TableHeaderCell className="text-center">
                ACCIONES
              </TableHeaderCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {categorias.length > 0 &&
              categorias.map((item) => (
                <TableRow key={item.id}>
                  <TableCell className="text-center">{item.id}</TableCell>
                  <TableCell>
                    <p
                      className="text-center"
                      id="td-nombre-categoria"
                    >
                      {item.nombre}
                    </p>
                  </TableCell>
                  <TableCell>
                    <p
                      data-cy="td-descripcion-categoria"
                      id="td-descripcion-categoria"
                    >
                      {item.descripcion}
                    </p>
                  </TableCell>
                  <TableCell className="flex gap-3">
                    <Link
                      className="text-white bg-blue-500 hover:bg-blue-600 px-4 py-2 rounded btn-editar"
                      to={`/categorias/${item.id}`}
                    >
                      Editar
                    </Link>
                    <button
                      className="text-white bg-red-500 hover:bg-red-600 px-4 py-2 rounded"
                      onClick={() => handleDelete(item.id)}
                    >
                      Eliminar
                    </button>
                  </TableCell>
                </TableRow>
              ))}
          </TableBody>
        </Table>
      </Card>
    </div>
  );
};
