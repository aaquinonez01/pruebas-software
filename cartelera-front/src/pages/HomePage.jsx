export const HomePage = () => {
  return (
    <div className="grid place-items-center w-full h-full ">
      <h1 className="text-center text-3xl">
        BIENVENIDO AL SISTEMA DE CARTELERA
      </h1>
    </div>
  );
};
