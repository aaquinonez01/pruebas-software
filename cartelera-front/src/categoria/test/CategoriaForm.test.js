import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { CategoriaForm } from '../pages/CategoriaForm';

describe('Test: <CategoriaForm/> ', () => {

    test('debe de mostrar el componente correctamente', () => {
        const wrapper = render(<MemoryRouter>
            <CategoriaForm />
        </MemoryRouter>);
        expect(wrapper).toMatchSnapshot();
    })

    test('debe los inputs estar vacios', () => {
        const { getByLabelText  } = render(<MemoryRouter>
            <CategoriaForm />
        </MemoryRouter>);
        const nombreInput = getByLabelText('Nombre');
        const descripcionInput = getByLabelText('Descripci贸n');
        expect(nombreInput.value).toBe('');
        expect(descripcionInput.value).toBe('');
    })
    
    test('debe de actualizar el estado cuando el input values cambie', () => {
        const { getByLabelText  } = render(<MemoryRouter>
            <CategoriaForm />
        </MemoryRouter>);
        const nombreInput = getByLabelText('Nombre');
        const descripcionInput = getByLabelText('Descripci贸n');
        fireEvent.change(nombreInput, { target: { value: 'Nuevo nombre' } });
         fireEvent.change(descripcionInput, { target: { value: 'Nueva descripci贸n' } });

        expect(nombreInput.value).toBe('Nuevo nombre');
        expect(descripcionInput.value).toBe('Nueva descripci贸n');
     })


});
