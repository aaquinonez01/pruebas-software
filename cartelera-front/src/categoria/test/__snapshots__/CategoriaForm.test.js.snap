// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test: <CategoriaForm/>  debe de mostrar el componente correctamente 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="tremor-Card-root relative w-full text-left ring-1 rounded-tremor-default bg-tremor-background ring-tremor-ring shadow-tremor-card dark:bg-dark-tremor-background dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card border-tremor-brand dark:border-dark-tremor-brand p-6"
        >
          <p
            class="font-medium text-tremor-title text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis text-center text-3xl"
          >
            Crear Nueva Categoría
          </p>
          <form
            class="mt-5 space-y-4 w-2/5 mx-auto"
          >
            <div>
              <label
                class="block font-medium"
                for="Nombre"
              >
                Nombre
              </label>
              <div
                class="tremor-TextInput-root relative flex items-center min-w-[10rem] outline-none rounded-tremor-default shadow-tremor-input dark:shadow-dark-tremor-input bg-tremor-background dark:bg-dark-tremor-background hover:bg-tremor-background-muted dark:hover:bg-dark-tremor-background-muted text-tremor-content dark:text-dark-tremor-content border-tremor-border dark:border-dark-tremor-border border w-full"
              >
                <input
                  class="tremor-TextInput-input w-full focus:outline-none focus:ring-0 border-none bg-transparent text-tremor-default text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis [appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none pl-4 pr-4 py-2 placeholder:text-tremor-content dark:placeholder:text-dark-tremor-content"
                  data-cy="nombre-categoria"
                  data-testid="base-input"
                  id="Nombre"
                  name="nombre"
                  placeholder="Ingrese el nombre de la categoría"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div>
              <label
                class="block font-medium"
                for="Descripcion"
              >
                Descripción
              </label>
              <div
                class="tremor-TextInput-root relative flex items-center min-w-[10rem] outline-none rounded-tremor-default shadow-tremor-input dark:shadow-dark-tremor-input bg-tremor-background dark:bg-dark-tremor-background hover:bg-tremor-background-muted dark:hover:bg-dark-tremor-background-muted text-tremor-content dark:text-dark-tremor-content border-tremor-border dark:border-dark-tremor-border border w-full"
              >
                <input
                  class="tremor-TextInput-input w-full focus:outline-none focus:ring-0 border-none bg-transparent text-tremor-default text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis [appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none pl-4 pr-4 py-2 placeholder:text-tremor-content dark:placeholder:text-dark-tremor-content"
                  data-cy="descripcion-categoria"
                  data-testid="base-input"
                  id="Descripcion"
                  name="descripcion"
                  placeholder="Ingrese la descripción de la categoría"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="flex justify-end"
            >
              <button
                class="tremor-Button-root flex-shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default shadow-tremor-input dark:shadow-dark-tremor-input border text-sm dark:bg-dark-tremor-brand border-tremor-brand dark:border-dark-tremor-brand dark:text-dark-tremor-brand-inverted dark:hover:bg-dark-tremor-brand-emphasis hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis bg-blue-500 hover:bg-blue-600 text-white px-4 py-2 rounded"
                data-cy="boton-actualizar"
              >
                <span
                  class="tremor-Button-text text-sm whitespace-nowrap"
                >
                  Crear Categoría
                </span>
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="tremor-Card-root relative w-full text-left ring-1 rounded-tremor-default bg-tremor-background ring-tremor-ring shadow-tremor-card dark:bg-dark-tremor-background dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card border-tremor-brand dark:border-dark-tremor-brand p-6"
      >
        <p
          class="font-medium text-tremor-title text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis text-center text-3xl"
        >
          Crear Nueva Categoría
        </p>
        <form
          class="mt-5 space-y-4 w-2/5 mx-auto"
        >
          <div>
            <label
              class="block font-medium"
              for="Nombre"
            >
              Nombre
            </label>
            <div
              class="tremor-TextInput-root relative flex items-center min-w-[10rem] outline-none rounded-tremor-default shadow-tremor-input dark:shadow-dark-tremor-input bg-tremor-background dark:bg-dark-tremor-background hover:bg-tremor-background-muted dark:hover:bg-dark-tremor-background-muted text-tremor-content dark:text-dark-tremor-content border-tremor-border dark:border-dark-tremor-border border w-full"
            >
              <input
                class="tremor-TextInput-input w-full focus:outline-none focus:ring-0 border-none bg-transparent text-tremor-default text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis [appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none pl-4 pr-4 py-2 placeholder:text-tremor-content dark:placeholder:text-dark-tremor-content"
                data-cy="nombre-categoria"
                data-testid="base-input"
                id="Nombre"
                name="nombre"
                placeholder="Ingrese el nombre de la categoría"
                type="text"
                value=""
              />
            </div>
          </div>
          <div>
            <label
              class="block font-medium"
              for="Descripcion"
            >
              Descripción
            </label>
            <div
              class="tremor-TextInput-root relative flex items-center min-w-[10rem] outline-none rounded-tremor-default shadow-tremor-input dark:shadow-dark-tremor-input bg-tremor-background dark:bg-dark-tremor-background hover:bg-tremor-background-muted dark:hover:bg-dark-tremor-background-muted text-tremor-content dark:text-dark-tremor-content border-tremor-border dark:border-dark-tremor-border border w-full"
            >
              <input
                class="tremor-TextInput-input w-full focus:outline-none focus:ring-0 border-none bg-transparent text-tremor-default text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis [appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none pl-4 pr-4 py-2 placeholder:text-tremor-content dark:placeholder:text-dark-tremor-content"
                data-cy="descripcion-categoria"
                data-testid="base-input"
                id="Descripcion"
                name="descripcion"
                placeholder="Ingrese la descripción de la categoría"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="flex justify-end"
          >
            <button
              class="tremor-Button-root flex-shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default shadow-tremor-input dark:shadow-dark-tremor-input border text-sm dark:bg-dark-tremor-brand border-tremor-brand dark:border-dark-tremor-brand dark:text-dark-tremor-brand-inverted dark:hover:bg-dark-tremor-brand-emphasis hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis bg-blue-500 hover:bg-blue-600 text-white px-4 py-2 rounded"
              data-cy="boton-actualizar"
            >
              <span
                class="tremor-Button-text text-sm whitespace-nowrap"
              >
                Crear Categoría
              </span>
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
